tests/Valkey.Glide.IntegrationTests/ReadFromEndToEndIntegrationTests.cs

```diff
@@ -0,0 +1,992 @@
+        var connectionMultiplexer = await ConnectionMultiplexer.ConnectAsync(configOptions);
+
+        // Assert - Verify connection was created successfully
+        Assert.NotNull(connectionMultiplexer);
+
+        // Verify ReadFrom configuration is set correctly
+        Assert.NotNull(configOptions.ReadFrom);
+        Assert.Equal(strategy, configOptions.ReadFrom.Value.Strategy);
+        Assert.Equal(az, configOptions.ReadFrom.Value.Az);
+
+        // Test a basic operation to ensure the connection works with ReadFrom configuration
+        var database = connectionMultiplexer.GetDatabase();
+        await database.PingAsync();
+
+        // Test data operations to verify the ReadFrom configuration is active
+        string testKey = Guid.NewGuid().ToString();
+        string testValue = "cluster-config-options-test";
+        await database.StringSetAsync(testKey, testValue);
+        string? retrievedValue = await database.StringGetAsync(testKey);
+        Assert.Equal(testValue, retrievedValue);
+
+        // Cleanup
+        await database.KeyDeleteAsync(testKey);
+        connectionMultiplexer.Dispose();
+    }
+
+    [Fact]
+    public async Task EndToEnd_ConfigurationOptions_NullReadFrom_DefaultsToNullInFFILayer()
+    {
+        // Arrange
+        var configOptions = new ConfigurationOptions
+        {
+            ReadFrom = null
+        };
+        configOptions.EndPoints.Add(TestConfiguration.STANDALONE_HOSTS[0].host, TestConfiguration.STANDALONE_HOSTS[0].port);
+        configOptions.Ssl = TestConfiguration.TLS;
+
+        // Act
+        var connectionMultiplexer = await ConnectionMultiplexer.ConnectAsync(configOptions);
+
+        // Assert - Verify connection was created successfully
+        Assert.NotNull(connectionMultiplexer);
+
+        // Verify ReadFrom is null (default behavior)
+        Assert.Null(configOptions.ReadFrom);
+
+        // Test a basic operation to ensure the connection works without ReadFrom configuration
+        var database = connectionMultiplexer.GetDatabase();
+        await database.PingAsync();
+
+        // Test data operations to verify default behavior works
+        string testKey = Guid.NewGuid().ToString();
+        string testValue = "null-readfrom-test";
+        await database.StringSetAsync(testKey, testValue);
+        string? retrievedValue = await database.StringGetAsync(testKey);
+        Assert.Equal(testValue, retrievedValue);
+
+        // Cleanup
+        await database.KeyDeleteAsync(testKey);
+        connectionMultiplexer.Dispose();
+    }
+
+    #endregion
+
+    #region Round-Trip Serialization Tests
+
+    [Theory]
+    [InlineData(ReadFromStrategy.Primary, null)]
+    [InlineData(ReadFromStrategy.PreferReplica, null)]
+    [InlineData(ReadFromStrategy.AzAffinity, "us-east-1a")]
+    [InlineData(ReadFromStrategy.AzAffinityReplicasAndPrimary, "eu-west-1b")]
+    public async Task EndToEnd_RoundTripSerialization_ParseToStringToParse_MaintainsConfigurationIntegrity(
```

Comment:
This test repeats EndToEnd_ConnectionString_ReadFromConfigurationFlowsToFFILayer_Standalone. What is the difference?